cdist/exec/remote.py

```diff
@@ -137,15 +137,14 @@ def extract_archive(self, path, mode):
         import cdist.autil as autil
 
         self.log.trace("Remote extract archive: %s", path)
-        # AIX only allows -C at the end
-        command = [
-            "tar",
-            "-x",
-            "-f", path,
-            "-C", os.path.dirname(path)
-        ]
+        opts = "xf"
         if mode is not None:
-            command += mode.extract_opts
+            opts += mode.extract_opts
```

Comment:
I'm currently testing this PR with many VMs and I've discovered some problems with BusyBox if the file name does not immediately follow the `f` option, i.e. `f` is not last.

Maybe remove `f` from `opts` and then append `f` in the `command` assignment below?